### PR TITLE
replace IndyHandle by separate types e.g. WalletHandle

### PR DESCRIPTION
**Pull request by AxelNennker**
Signed-off-by: Axel Nennker <axel.nennker@telekom.de>

Created types for each kind of handle e.g. WalletHandle, CommandHandle, ...

Also moved the type definition of IndyHandle in https://github.com/hyperledger/indy-sdk/blob/master/wrappers/rust/src/lib.rs#L40 to https://github.com/hyperledger/indy-sdk/blob/master/wrappers/rust/indy-sys/src/lib.rs because all types used in the api should be defined in indy-sys

To achieve real type safety more work needs to be done

### CODE REVIEW

File: wrappers/rust/src/blob_storage.rs [hunk 1]
```diff
@@ -1,38 +1,39 @@
...
 use std::ffi::CString;
 
 use ffi::blob_storage;
 use ffi::ResponseI32CB;
 
 use utils::callbacks::{ClosureHandler, ResultHandler};
+use ffi::CommandHandle;
 
-pub fn open_reader(xtype: &str, config_json: &str) -> Box<Future<Item=IndyHandle, Error=IndyError>> {
+pub fn open_reader(xtype: &str, config_json: &str) -> Box<Future<Item=CommandHandle, Error=IndyError>> {
```
Comment (jovfer) [inline wrappers/rust/src/blob_storage.rs]: Is it really CommandHandle in the result? AFAIR it should be blob reader handle, but I may be wrong

Comment (AxelNennker) [reply wrappers/rust/src/blob_storage.rs]: actually that seems to be something I don't have yet. It is a ConfigHandle.
https://github.com/hyperledger/indy-sdk/blob/master/libindy/src/services/blob_storage/mod.rs#L128

Comment (AxelNennker) [reply wrappers/rust/src/blob_storage.rs]: It seems I would have to add something like ResponseI32CB for each new type that replaces i32 e.g. ResponseBlobStorageConfigHandleCB. Which is a lot of work. Probably this should be done type by type.
Some more macros might help. Maybe reintroduce IndyHandle for now?

The amount of boilerplate in indy-sdk is quite overwhelming.

Comment (jovfer) [reply wrappers/rust/src/blob_storage.rs]: I think temporary usage of IndyHandle until we will have all separates type is better, rather then using CommandHandle as result type for opened blob reader.

**Pull request merged**